topic_dynamics/parsing.py

```diff
@@ -0,0 +1,420 @@
+"""
+Parsing-related functionality.
+"""
+
+from collections import Counter, namedtuple
+import datetime
+from operator import itemgetter
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Any, List, Tuple
+
+from tqdm import tqdm
+import tree_sitter
+
+from .parsers.utils import get_parser
+from .slicing import get_dates, checkout_by_date
+
+NODE_TYPES = {"c": ["identifier", "type_identifier"],
+              "c-sharp": ["identifier", "type_identifier"],
+              "cpp": ["identifier", "type_identifier"],
+              "java": ["identifier", "type_identifier"],
+              "python": ["identifier", "type_identifier"]}
+
+SliceLine = namedtuple("SliceLine", "date start_index end_index")
```

Comment:
Please use [type hints everywhere](https://stackoverflow.com/questions/34269772/type-hints-in-namedtuple)